python/tests/array_metadata.py

```diff
@@ -14,49 +14,64 @@ def test_array_metadata(self):
         dpml = 2
 
         sxy = 2*(r+w+pad+dpml)
+        cell_size = mp.Vector3(sxy,sxy)
+
         nonpml_vol = mp.Volume(mp.Vector3(), size=mp.Vector3(sxy-2*dpml,sxy-2*dpml))
 
-        c1 = mp.Cylinder(radius=r+w, material=mp.Medium(index=n))
-        c2 = mp.Cylinder(radius=r)
+        geometry = [mp.Cylinder(radius=r+w, material=mp.Medium(index=n)),
+                    mp.Cylinder(radius=r)]
 
         fcen = 0.118
         df = 0.08
 
+        symmetries = [mp.Mirror(mp.X,phase=-1),
+                      mp.Mirror(mp.Y,phase=+1)]
+
+        pml_layers = [mp.PML(dpml)]
+
         # CW source
-        src = mp.Source(mp.ContinuousSource(fcen,fwidth=df), mp.Ez, mp.Vector3(r+0.1))
+        src = [mp.Source(mp.ContinuousSource(fcen,fwidth=df), mp.Ez, mp.Vector3(r+0.1)),
+               mp.Source(mp.ContinuousSource(fcen,fwidth=df), mp.Ez, mp.Vector3(-(r+0.1)), amplitude=-1)]
 
-        sim = mp.Simulation(cell_size=mp.Vector3(sxy,sxy),
-                            geometry=[c1,c2],
-                            sources=[src],
+        sim = mp.Simulation(cell_size=cell_size,
+                            geometry=geometry,
+                            sources=src,
                             resolution=resolution,
                             force_complex_fields=True,
-                            symmetries=[mp.Mirror(mp.Y)],
-                            boundary_layers=[mp.PML(dpml)])
+                            symmetries=symmetries,
+                            boundary_layers=pml_layers)
 
         sim.init_sim()
         sim.solve_cw(1e-6, 1000, 10)
-        cw_modal_volume = sim.modal_volume_in_box(box=nonpml_vol)
+
+        def electric_energy(r, ex, ey, ez, eps):
+            return np.real(eps * (np.conj(ex)*ex + np.conj(ey)*ey + np.conj(ez)*ez))
```

Comment:
To make sure the `X`, `Y`, `Z` arrays are correct, see my suggestion below about multiplying by a function of `r` too.

(We don't need to test this on a physically meaningful integrand.)